ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/workspace/WorkspaceComponent.java

```diff
@@ -166,55 +167,58 @@ public void setCurrentWorkspace(Workspace workspace) {
      */
     public void handleWorkspaceEvents(final WorkspaceDto workspace, final Callback<Component, Exception> callback,
                                       final Boolean restoreFromSnapshot) {
-        this.callback = callback;
-        if (messageBus != null) {
-            messageBus.cancelReconnection();
+
+        loader.show(STARTING_WORKSPACE_RUNTIME);
+
+        setCurrentWorkspace(workspace);
+
+        String workspaceId = appContext.getWorkspaceId();
+
+        subscribe(WS_STATUS_ERROR_MSG, "event:workspace-status:subscribe", workspaceId);
+        subscribe(WS_AGENT_OUTPUT_ERROR_MSG, "event:ws-agent-output:subscribe", workspaceId);
```

Comment:
Well, I believe there is not much we can do here because in any way we have to subscribe for those events (workspace-status, workspace-agent-output, etc.). It does not really matter what way we choose we will do it each time for each event type. If I remember well, these are not a REST methods but JSON RPC, I'll report back if I'm mistaken.